inc/class-s3-uploads-stream-wrapper.php

```diff
@@ -17,7 +17,35 @@ public static function register( Aws\S3\S3Client $client)
 		static::$client = $client;
 	}
 
+	// Override
+	public function stream_flush() {
+
+		// Add some HTTP headers which can be defined in wp-config.php
+
+		// Expires:
+		if ( defined( 'S3_UPLOADS_EXPIRES' ) ) {
+			$this->params[ 'Expires' ] = S3_UPLOADS_EXPIRES;
+		}
+
+		// Cache-Control:
+		if ( defined( 'S3_UPLOADS_CACHE_CONTROL' ) ) {
```

Comment:
Ideally these constant should be passed into the constructor, so the caller controls whether to use the constants for not, this way the constants are decoupled form the class. It's possible someone would want to use their own instance of this class without the global constants setting these options.

See https://github.com/humanmade/S3-Uploads/blob/master/inc/class-s3-uploads.php#L20 for a similar example - the Shared Instance uses the wp-config constants, but it's still possible to initiate a new instance with your own details if you want. Though we don't have  a shared instance here - a similar logic can be applied.